### PR TITLE
Fix types in napari/layers/utils/text_manager.py, and string_encoding.py

### PR DESCRIPTION
**Pull request by Carreau**
I'm not quite sure what was meant as Array was not generic. Ane the Literal[Tuples] are a bit handwaved , it seem like Literal[-1] and Literal[(-1,)] might be that same instead of actualy checing for tuple, at least I'm quite unsure about how this behave.

### CODE REVIEW

File: napari/layers/utils/string_encoding.py [hunk 1]
```diff
@@ -14,14 +23,14 @@
...
+StringValue = Array[str, Tuple[()]]
 
 """An Nx1 array where each element represents one string value."""
-StringArray = Array[str, (-1,)]
+StringArray = Array[str, Literal[(-1,)]]
 
 
 """The default string value, which may also be used a safe fallback string."""
-DEFAULT_STRING = np.array('', dtype='<U1')
+DEFAULT_STRING = cast(Array[str, Tuple[()]], np.array('', dtype='<U1'))
```
Comment (Carreau) [inline napari/layers/utils/string_encoding.py:33]: @andy-sweet I think you are the one who wrote this.

**Comment by andy-sweet** [comment]
I'm planning to get to this, but have been busy with other things recently. Should be able to review this and likely suggest some changes early next week.

**Comment by andy-sweet** [comment]
As the PR stands, I don't think it helps most readers clarify what types are expected here.

For context, I think that `Array` was defined as a custom type that could be used in a [Pydantic model](https://docs.pydantic.dev/1.10/usage/types/#classes-with-__get_validators__) to automatically validate an input that could be coerced to a numpy array based on the provided dtype and optional shape.

The definition is a little strange though since calling the validator (`Array.validate_type`) returns an `np.ndarray` and not an `Array`, and I think that causes some typing confusion.

I originally defined the `StringValue` and `StringArray` classes to use that convenient definition and to be consistent with `ColorValue` and `ColorArray`. Since then, we updated the `ColorValue` and `ColorArray` validators to return a view over the numpy array as their respective types, which helped clarify some typing usage.

So I have a few ideas in this PR.

1\. Change `Array` to be like [`ColorValue`](https://github.com/napari/napari/blob/591809d6c3194dc381292d5353b3936ed8e7f0c8/napari/utils/color.py#L74), so that the validator returns an array of type `Array`.

This has the widest impact. The main downside is that when inspecting attribute types, users will see napari's custom `Array` instead of `np.ndarray`. Given the spirit of duck typing in Python, maybe this is fine?

2\. Define `StringValue` and `StringArray` like `ColorValue` and `ColorArray`.

This is the simplest change that would help get this PR through. But it seems a bit silly to keep define these special types.

3\. Deprecate these types and just use functions for validation. 

I think the only downside here is a little more boilerplate code, but the upside is simplicity.

What do you think?

**Comment by Carreau** [comment]
> As the PR stands, I don't think it helps most readers clarify what types are expected here, even if mypy is now happy

Yeah, I was not trying to make things clear anymore, I hope we can first get mypy happy on all files without too much fuss and then refine.

I'm all for uniformity and personally I find Pydantic a bit too complex / not worth the usage unless we want serialization, so I'm leaning for 2 or 3.


> This is the simplest change that would help get this PR through. But it seems a bit silly to keep define these special types.

I think we cal always rely on NewType pattern to make this affects only type checking.

**Pull request closed without merging**